src/pkg/shub/client/util_test.go

```diff
@@ -0,0 +1,63 @@
+// Copyright (c) 2018, Sylabs Inc. All rights reserved.
+// This software is licensed under a 3-clause BSD license. Please consult the
+// LICENSE.md file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+package client
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/singularityware/singularity/src/pkg/test"
+)
+
+// TestShubPacker checks if we can create a Bundle from the pulled image
+func TestShubParser(t *testing.T) {
+	test.DropPrivilege(t)
+	defer test.ResetPrivilege(t)
+
+	validShubURIs := []string{
+		`//username/container`,
+		`//username/container:tag`,
+		`//username/container@00000000000000000000000000000000`,
```

Comment:
What makes this URI valid but makes the URI on line 37 invalid? I can't tell the difference between the two

Reply:
The digest on this one is 32 digits while the one on 37 is 31 digits and the one on 38 is 33 digits